test/e2e/util.go

```diff
@@ -182,3 +186,117 @@ func randomSuffix() string {
 func expectNoError(err error, explain ...interface{}) {
 	ExpectWithOffset(1, err).NotTo(HaveOccurred(), explain...)
 }
+
+func cleanup(filePath string, selectors ...string) {
+	By("using stop to clean up resources")
+	runKubectl("stop", "-f", filePath)
+
+	for _, selector := range selectors {
+		resources := runKubectl("get", "pods,rc,se", "-l", selector, "--no-headers")
+		if resources != "" {
+			Failf("Resources left running after stop:\n%s", resources)
+		}
+	}
+}
+
+// validatorFn is the function which is individual tests will implement.
+// we may want it to return more than just an error, at some point.
+type validatorFn func(c *client.Client, podID string) error
+
+// validateController is a generic mechanism for testing RC's that are running.
+// It takes a container name, a test name, and a validator function which is plugged in by a specific test.
+// "containername": this is grepped for.
+// "testname":  which gets bubbled up to the logging/failure messages if errors happen.
+// "validator" function: This function is given a podID and a client, and it can do some specific validations that way.
+func validateController(c *client.Client, image string, replicas int, containername string, testname string, validator validatorFn) {
+	getPodsTemplate := "--template={{range.items}}{{.id}} {{end}}"
+	// NB: kubectl adds the "exists" function to the standard template functions.
+	// This lets us check to see if the "running" entry exists for each of the containers
+	// we care about. Exists will never return an error and it's safe to check a chain of
+	// things, any one of which may not exist. In the below template, all of info,
+	// containername, and running might be nil, so the normal index function isn't very
+	// helpful.
+	// This template is unit-tested in kubectl, so if you change it, update the unit test.
+	// You can read about the syntax here: http://golang.org/pkg/text/template/.
+	containername = "update-demo"
```

Comment:
Still need to delete this line, otherwise the containername arg isn't doing anything.